### ISSUE TITLE
[Daily 9/11] entire body recreated when using up to face to a new body on protrusion

### ISSUE THREAD

**Issue by lunaticstraydog** [realthunder/FreeCAD_assembly3#810]
Steps to reproduce:
- create a shape (cube for instance) in one body
- create another shape in another body
- create a sketch onto the first body , and protrude it up to the second body (using binder auto creation implicitely)  
- You will end up with something like that:  
![image](https://user-images.githubusercontent.com/25230216/132977086-dc10b78c-7ba3-473d-b46f-2fbc0a301d10.png)
As compared to when setting a simple dimension :
![image](https://user-images.githubusercontent.com/25230216/132977177-070844b8-6d87-4a8e-bbc8-dbec80af436e.png)
Basically, the first shape, here the cube, is considered part of the protrusion .  
That means that a mirror operation for instance will mirror all of the previous shapes.  
<details>
  <summary>FreeCAD info:</summary>
  OS: Arch Linux (sway)
Word size of OS: 64-bit
Word size of FreeCAD: 64-bit
Version: 2021.911.24301 +4130 (Git) AppImage
Build type: Release
Branch: LinkDaily
Hash: cdcdcfc78188218bd5eab1cfd9f911acd666f734
Python version: 3.9.7
Qt version: 5.12.9
Coin version: 4.0.1
OCC version: 7.5.2
Locale: French/France (fr_FR)  
</details>

**Comment by mwlandolsi** [comment]
confirmed,
Mirrored extruded shape only, causes the mirro of the base shape, which is not selected.
![Screenshot from 2021-09-13 13-54-47](https://user-images.githubusercontent.com/76944804/133087373-26fbe162-1088-46bb-8b4a-5dba7e02fa9b.png)

OS: Linux Mint 20.1 (X-Cinnamon/cinnamon)
Word size of OS: 64-bit
Word size of FreeCAD: 64-bit
Version: 2021.911.24301 +4130 (Git) AppImage
Build type: Release
Branch: LinkDaily
Hash: cdcdcfc78188218bd5eab1cfd9f911acd666f734
Python version: 3.9.7
Qt version: 5.12.9
Coin version: 4.0.1
OCC version: 7.5.2
Locale: English/United States (en_US)
